### PR TITLE
[SYCL][Graph][Doc] Tidy of graph extension design doc

### PR DESCRIPTION
**Pull request by EwanC**
Cleans up the design doc a bit with the following changes:

* Reflow lines to 80 chars.
* Fix some typos and change usages of UK English to US.
* Update intro that talks about emulation mode and only supporting L0.
* Add section on resources at the beginning where we can link public information on the extension we've created.

### CODE REVIEW

File: sycl/doc/design/CommandGraph.md [hunk 1]
```diff
@@ -15,13 +19,11 @@ represents a series of operations to be enqueued to the backend device and
 their dependencies. A single command-graph can be partitioned into more
 than one command-buffer by the runtime. The SYCL Graph extension
 distinguishes between backends that support the command-buffer extension
-and those that do not. Currently command-buffer extensions are only supported
-by Level Zero. All other backends would fall back to an emulation mode, or not
-be reported as supported.
+and those that do not, and only reports support for the extension using an
```
Comment (fabiomestre) [inline sycl/doc/design/CommandGraph.md:22]: what does "using an aspect" mean in this context?

Comment (EwanC) [reply sycl/doc/design/CommandGraph.md]: Good question, it refers to the `ext_oneapi_graph` aspect [defined by our extension](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc#73-querying-device-support), so i've made that explicit.

See https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:device-aspects for the definition of SYCL aspects.

Comment (fabiomestre) [reply sycl/doc/design/CommandGraph.md]: Thanks for clarifying. The update looks good to me :+1:

**Review by fabiomestre — APPROVED** [review]

**Comment by EwanC** [comment]
@intel/llvm-gatekeepers This is good to go, thanks

**Pull request merged**